core/pkg/artifacts/downloader.go

```diff
@@ -84,10 +117,49 @@ func (ad *ArtifactDownloader) getArtifactManifest(artifactID string) (manifest M
 		return Manifest{}, fmt.Errorf("could not access manifest for artifact")
 	}
 	directURL := artifactManifest.GetFile().DirectUrl
-	manifest, err = loadManifestFromURL(directURL)
+
+	// Download and decode
+	return ad.downloadManifestFromURL(directURL)
+}
+
+func (ad *ArtifactDownloader) downloadManifestFromURL(
+	url string,
+) (Manifest, error) {
+	req, err := retryablehttp.NewRequest(http.MethodGet, url, nil)
 	if err != nil {
-		return Manifest{}, err
+		return Manifest{}, fmt.Errorf("create request failed: %w", err)
```

Comment:
Use the `%v` verb instead of `%w` if the errors are not unwrapped. If you do use the `%w` verb, the function must document what error types are expected.